spec/classes/server_spec.rb

```diff
@@ -367,6 +367,34 @@
         it { is_expected.to contain_file('/etc/zabbix/zabbix_server.conf').without_content %r{^StartODBCPollers=1} }
       end
 
+      context 'with zabbix_server.conf and version 7.0' do
+        next if facts[:os]['name'] == 'Debian' && facts[:os]['release']['major'].to_i <= 12
+        next if facts[:os]['name'] == 'Ubuntu' && facts[:os]['release']['major'].to_i <= 20
+
+        let :params do
+          {
+            smsdevices: ['/dev/ttyS0'],
+            zabbix_version: '7.0'
+          }
+        end
+
+        it { is_expected.to contain_file('/etc/zabbix/zabbix_server.conf').with_content %r{^SMSDevices=/dev/ttyS0} }
+      end
+
+      context 'with zabbix_server.conf and version 7.2' do
+        next if facts[:os]['name'] == 'Debian' && facts[:os]['release']['major'].to_i <= 12
```

Comment:
Please see my comment in https://github.com/voxpupuli/puppet-zabbix/pull/991#discussion_r1987044848 . the OS check shouldn't be in the tests. The zabbix version check needs to be in the template. otherwise we will get a broken config on zabbix 6.x and older and the tests will mask that.